bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java

```diff
@@ -19,32 +20,94 @@
+
+    public Http4kSlackApp(App app, String path) {
         this.app = app;
+        this.path = path;
         requestParser = new SlackRequestParser(app.config());
     }
 
     @Override
     public Response invoke(Request request) {
+        Method requestMethod = request.getMethod();
+        String requestPath = request.getUri().getPath();
+        if (requestMethod == null) {
+            // checking this, just in case
+            return notFound(request);
+        }
+        if (requestMethod == Method.POST && requestPath.equals(getSlackEventsRequestURI())) {
+            return handleEventRequest(request);
+        } else if (requestMethod == Method.GET) {
+            if (app.config().isOAuthInstallPathEnabled()
+                    && requestPath.equals(app.config().getOauthInstallRequestURI())) {
+                return handleOAuthRequest(request);
+            } else if (app.config().isOAuthInstallPathEnabled()
+                    && requestPath.equals(app.config().getOauthRedirectUriRequestURI())) {
+                return handleOAuthRequest(request);
+            }
+        }
+        return notFound(request);
+    }
+
+    // ----------------------------------------------------------
+
+    protected String getSlackEventsRequestURI() {
+        if (app.config().getAppPath() != null) {
+            return (app.config().getAppPath() + this.path).replaceAll("//", "/");
+        }
+        return this.path;
+    }
+
+    protected Response handleEventRequest(Request request) {
         try {
             return toHttp4kResponse(app.run(toSlackRequest(request)));
         } catch (Exception e) {
             LOGGER.error(e.getMessage(), e);
-            return Response.Companion.create(INTERNAL_SERVER_ERROR)
-                    .header("Content-type", "application/json; charset=utf-8")
-                    .body("{\"error\":\"An error occurred during processing of the request\"}");
+            return buildEventErrorResponse(request, e);
         }
     }
 
-    private com.slack.api.bolt.request.Request<?> toSlackRequest(Request httpRequest) {
+    protected Response handleOAuthRequest(Request request) {
+        try {
+            return toHttp4kResponse(app.run(toSlackRequest(request)));
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+            return buildOAuthPageErrorResponse(request, e);
+        }
+    }
+
+    protected Response buildEventErrorResponse(Request request, Exception e) {
+        return Response.Companion.create(INTERNAL_SERVER_ERROR)
+                .header("Content-type", "application/json; charset=utf-8")
+                .body("{\"error\":\"An error occurred during processing of the request\"}");
+    }
+
+    protected Response buildOAuthPageErrorResponse(Request request, Exception e) {
+        return Response.Companion.create(INTERNAL_SERVER_ERROR)
+                .header("Content-type", "text/plain; charset=utf-8")
+                .body("Internal Server Error");
```

Comment:
Regarding the default end-user-facing pages and some other OAuth flow behaviors, I'd like to align with Bolt for Python in Java SDK v2.